api/helpers/email.js

```diff
@@ -109,33 +113,23 @@ exports.sendWelcomeEmail = async function (projectName, email) {
 
     // Send the emails to the CHES (Common Hosted Email Service)
     try {
-        const getOKRes = await axios.post(_GETOK_endpoint,
-            qs.stringify({
-                'client_id': _GETOK_CLIENTID,
-                'client_secret': _GETOK_CLIENT_SECRET,
-                'grant_type': 'client_credentials',
-            }),
-            {
-                headers: {
-                    'content-type': 'application/x-www-form-urlencoded'
-                }
-            }
-        );
-        if (getOKRes && getOKRes.data && getOKRes.data.access_token) {
+        const emailToken = await getEmailToken();
+
+        if (emailToken && emailToken.data && emailToken.data.access_token) {
             // Send the welcome email
             await axios.post(
```

Comment:
isn't line 71 axios call same as this line, wondering if we could combine these calls, similar to what you did with the getEmailToken function

Reply:
After looking into this - I don't think there's an easy way to trim this further. Basically each email HTTP request requires an auth token. We currently request a token for each request before making it and as such we need to check that the token was successfully received beforehand.

Ideally we'd set up a request middleware that gets the token and saves it to memory(and refresh it as needed, upon expiry) so we don't to a token request each time we need to use the email API.